pkg/oonf_api/0001-add-RIOT-support.patch

```diff
@@ -116,145 +115,48 @@ index 103c88f..8a6cd2b 100644
 @@ -48,7 +48,7 @@
  #include "common/common_types.h"
  #include "common/daemonize.h"
- 
+
 -#ifndef WIN32
 +#if (!defined(_WIN32)) && (!defined(RIOT_VERSION))
  /**
   * Prepare the start of a daemon. Fork into background,
   * but keep stdin/out/err open and a pipe connected to
 diff --git a/src-api/common/netaddr.c b/src-api/common/netaddr.c
-index dedab2c..1b602ec 100644
+index dedab2c..6b214ee 100644
 --- a/src-api/common/netaddr.c
 +++ b/src-api/common/netaddr.c
-@@ -43,7 +43,14 @@
+@@ -43,7 +43,11 @@
  #include <stdio.h>
  #include <stdlib.h>
  #include <string.h>
 +#ifndef RIOT_VERSION
  #include <net/if.h>
 +#else
-+#include <arpa/inet.h>
-+#include "net/af.h"
 +#define DONT_HAVE_SIN6_SCOPE_ID
 +#endif
- 
+
  #include "common/common_types.h"
  #include "common/string.h"
-@@ -175,12 +182,12 @@ netaddr_to_binary(void *dst, const struct netaddr *src, size_t len) {
- int
- netaddr_from_socket(struct netaddr *dst, const union netaddr_socket *src) {
-   memset(dst->_addr, 0, sizeof(dst->_addr));
--  if (src->std.sa_family == AF_INET) {
-+  if (src->std.sin6_family == AF_INET) {
```

Comment:
tbh I don't quite get this line :/ If the address family is v4, assume src is a ``sockaddr_in6`` but then set its family to AF_INET instead of AF_INET6?